expression/builtin_json.go

```diff
@@ -772,6 +773,65 @@ type jsonLengthFunctionClass struct {
 	baseFunctionClass
 }
 
+type builtinJSONLengthSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinJSONLengthSig) Clone() builtinFunc {
+	newSig := &builtinJSONLengthSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
 func (c *jsonLengthFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_LENGTH")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	argTps := make([]types.EvalType, 0, len(args))
+	argTps = append(argTps, types.ETJson)
+	if len(args) == 2 {
+		argTps = append(argTps, types.ETString)
+	}
+
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, argTps...)
+	sig := &builtinJSONLengthSig{bf}
+	sig.setPbCode(tipb.ScalarFuncSig_JsonLengthSig)
+	return sig, nil
+}
+
+func (b *builtinJSONLengthSig) evalInt(row chunk.Row) (res int64, isNull bool, err error) {
+	obj, isNull, err := b.args[0].EvalJSON(b.ctx, row)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+
+	if obj.Type() != "OBJECT" && obj.Type() != "ARRAY" {
+		return 1, false, nil
+	}
+
+	if len(b.args) == 2 {
```

Comment:
how about:
```go
if len(b.args) == 1 {
	return int64(obj.GetElemCount()), false, nil
}
// handle the case that len(b.args) == 2
...
```

Reply:
actually, I'm ready to extract the common path visit but in a other pr(some other pr are conflicted with that)